File: MonsterHunterStories/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MonsterHunterStories
{
	internal class Item
	{
		private readonly uint mAddress;

		public Item(uint address)
		{
			mAddress = address;
		}

		public uint ID
		{
			get => SaveData.Instance().ReadNumber(mAddress, 2);
		}

		public uint Count
		{
			get => SaveData.Instance().ReadNumber(mAddress + 2, 2);
			set => Util.WriteNumber(mAddress, 2, 0, 99, value);
		}
	}
}

File: item.py
from save_data import SaveData
from util import Util

class Item:
    def __init__(self, address):
        self.address = address

    @property
    def id(self):
        return SaveData.instance().read_number(self.address, 2)

    @property
    def count(self):
        return SaveData.instance().read_number(self.address + 2, 2)

    @count.setter
    def count(self, value):
        Util.write_number(self.address + 2, 2, 0, 999, value)
